wire-ios-request-strategy/Sources/Message Sending/MessageSender.swift

```diff
@@ -0,0 +1,240 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension Swift.Result {
+    func flatMapAsync<NewSuccess>(_ transform: (Success) async -> Swift.Result<NewSuccess, Failure>) async -> Swift.Result<NewSuccess, Failure> {
+        switch self {
+        case .success(let success):
+            return await transform(success)
+        case .failure(let failure):
+            return .failure(failure)
+        }
+    }
+}
```

Comment:
move to WireUtilities?

Reply:
Will delete it after moving away from Result based error handling